src/runtime-rs/crates/runtimes/common/src/runtimes_utils/host_hooks.rs

```diff
@@ -0,0 +1,371 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::process::Stdio;
+use std::{collections::HashMap, path::PathBuf, time::Duration};
+
+use anyhow::{anyhow, Result};
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+
+logging::logger_with_subsystem!(sl, "runtimes");
+
+pub enum HookType {
+    PreStart,
+    CreateRuntime,
+    PostStart,
+    PostStop,
+}
+
+// execute hooks in oci::Hooks according to hook types
+pub async fn run_hooks(spec: &oci::Spec, st: &oci::State, hk_type: HookType) -> Result<()> {
+    if let Some(spec_hooks) = &spec.hooks {
+        let hooks = match hk_type {
+            HookType::PreStart => {
+                if !spec_hooks.prestart.is_empty() {
```

Comment:
I think this log and belows have no meaning, so delete them?

Reply:
We might need some log to indicate that the hook is running?